### PR TITLE
added emoji feature

### PR DESCRIPTION
**Pull request by tanay-pingalkar**
before
Screenshot from 2021-02-21 13-00-40
After
Screenshot from 2021-02-21 12-59-10
after on click the smile
Screenshot from 2021-02-21 12-59-44
after on select any of smile you want
Screenshot from 2021-02-21 13-03-54

Library I used : emoji-mart
The size of emoji-sheet is 1.34 MB. If you want to minimize the size you can do sheetsize to 16 but it affect on the quality of emoji.

### CODE REVIEW

**Comment by tanay-pingalkar** [comment]
not tested with two users because i was not able to create test user. But it will work because my code has nothing to do with any backend.

**Comment by nadirabbas** [comment]
It works.

**Comment by tanay-pingalkar** [comment]
> It works.

THANKS

**Comment by ajmnz** [comment]
Nice one @tanay-pingalkar. Should the emoji box be closed when the input is focused again?

**Comment by benawad** [comment]
I want the emoji button to be on the right and a little smaller

**Comment by tanay-pingalkar** [comment]
> Nice one @tanay-pingalkar. Should the emoji box be closed when the input is focused again?

but what if you wat to send multiple emojis

**Comment by tanay-pingalkar** [comment]
> Nice one @tanay-pingalkar. Should the emoji box be closed when the input is focused again?

no, but its easy to add, i will add..

**Comment by ajmnz** [comment]
Something like that should do it @tanay-pingalkar 

Captura de Pantalla 2021-02-21 a les 16 20 19

**Comment by tanay-pingalkar** [comment]
> Something like that should do it @tanay-pingalkar
> 
> Captura de Pantalla 2021-02-21 a les 16 20 19

how you made this
mine one is actually ugly.
i am also gonna change its color to gray.
do you have any resource from where i can get icon svgs. @ajmnz

**Comment by ajmnz** [comment]
> how you made this

Your emoji icon is absolutely positioned, so just put its `right` to some low value. Ideally, the input should have a wrapper with relative positioning, so everything can be arranged based on that. 

Let me know if you need any help with that.

**Comment by ajmnz** [comment]
> do you have any resource from where i can get icon svgs. @ajmnz

[Feather](https://feathericons.com/), [Boxicons](https://boxicons.com/), [Essential Icons](http://www.essential-icons.com/), [Linearicons](https://linearicons.com/), [Icomoon](https://icomoon.io/), and many more. Those are some of the packs that I currently use when designing UIs. Depends on the project tho

**Comment by benawad** [comment]
also have the emoji modal go above the input

**Comment by tanay-pingalkar** [comment]
> also have the emoji modal go above the input

I made it below because I thought it will look like whatsApp and telegram.

**Comment by nadirabbas** [comment]
@tanay-pingalkar the mentions show above too. so it's just consistent. And it doesn't make the actual input move from its place.

**Comment by tanay-pingalkar** [comment]
> @tanay-pingalkar the mentions show above too. so it's just consistent. And it doesn't make the actual input move from its place.

Oki

**Comment by ajmnz** [comment]
@tanay-pingalkar Let me know if you need help, I can jump in whenever you feel like

**Comment by tanay-pingalkar** [comment]
> @tanay-pingalkar Let me know if you need help, I can jump in whenever you feel like

Currently I am not able to restart elixir server

**Comment by nadirabbas** [comment]
Hit CTRL + C?

**Comment by tanay-pingalkar** [comment]
> Hit CTRL + C?





I had closed vs code
I will see that tomorrow

**Comment by tanay-pingalkar** [comment]
what is hex color of other icons in dogehouse.

**Comment by tanay-pingalkar** [comment]
should i make emoji component in different file

**Comment by tanay-pingalkar** [comment]
Now
Screenshot from 2021-02-22 13-09-35
Screenshot from 2021-02-22 13-09-56



> Nice one @tanay-pingalkar. Should the emoji box be closed when the input is focused again?

yup i added this feature

**Comment by ajmnz** [comment]
@tanay-pingalkar Can you add `cursor: pointer` to the emoji button?

**Comment by tanay-pingalkar** [comment]
> @tanay-pingalkar Can you add `cursor: pointer` to the emoji button?

it can be only possible if emoji-mart have that feature in there documentation

**Comment by ajmnz** [comment]
> it can be only possible if emoji-mart have that feature in there documentation

I'll make a PR in your fork

File: kofta/src/vscode-webview/modules/room-chat/RoomChatInput.tsx
Comment (ajmnz) [inline kofta/src/vscode-webview/modules/room-chat/RoomChatInput.tsx]: Add `cursor: pointer here`.

**Review by ajmnz — CHANGES_REQUESTED** [review]

File: kofta/src/vscode-webview/modules/room-chat/RoomChatInput.tsx
Comment (tanay-pingalkar) [reply kofta/src/vscode-webview/modules/room-chat/RoomChatInput.tsx]: ok

**Comment by ajmnz** [comment]
@tanay-pingalkar You can add it to the `i` element

**Comment by tanay-pingalkar** [comment]
> @tanay-pingalkar You can add it to the `i` element

i thought you told to add to emoji console

**Comment by tanay-pingalkar** [comment]
should i make emoji feature in different file?

File: kofta/src/vscode-webview/modules/room-chat/RoomChatInput.tsx
Comment (ajmnz) [inline kofta/src/vscode-webview/modules/room-chat/RoomChatInput.tsx]: This logic makes it so the emoji is always appended to the end of the message. In my opinion it should be placed wherever the cursor is. `inputRef.current.selectionEnd`

**Review by ajmnz — CHANGES_REQUESTED** [review]

File: kofta/src/vscode-webview/modules/room-chat/RoomChatInput.tsx
Comment (tanay-pingalkar) [reply kofta/src/vscode-webview/modules/room-chat/RoomChatInput.tsx]: ok

**Comment by tanay-pingalkar** [comment]
done anything else

**Comment by ajmnz** [comment]
Looks good to me, we'll wait for Ben's approval.

**Comment by tanay-pingalkar** [comment]
there is problem, the is is gonna added whenever the cursor is but whenever you click on emoji icon the cursor will be no longer visible. How the user will know where his cursor is. Do have any idea

**Comment by ajmnz** [comment]
> there is problem, the is is gonna added whenever the cursor is but whenever you click on emoji icon the cursor will be no longer visible. How the user will know where his cursor is. Do have any idea

I mean, the user intentionally moves the cursor wherever he/she wants the emoji to be. I don't think there's a way to maintain the input focused while picking an emoji. Focusing back after picking the emoji would improve the UX, which could be better than the close-on-focus feature.

Anyway, we'll wait for Ben's feedback.

**Comment by nadirabbas** [comment]
@ajmnz I could do something to maintain the focus.

**Comment by tanay-pingalkar** [comment]
i just fixed a very major bug.

**Comment by tanay-pingalkar** [comment]
what is the conflict

**Comment by tanay-pingalkar** [comment]
I am thinking to make all chat feature in seperate file. Like emoji feature, mention feature and all upcoming feature. Because it is getting too hard read the ChatAppInput.tsx .

**Review by DGKSK8LIFE — APPROVED** [review]

**Comment by tanay-pingalkar** [comment]
@benawad   emoji feature is ready now.

**Pull request merged**
